building_sim_plugins/building_ignition_plugins/src/crowd_simulator.cpp

```diff
@@ -302,6 +302,19 @@ void CrowdSimulatorPlugin::_config_spawned_agents(
     *animation_name_comp = ignition::gazebo::components::AnimationName(
       animation_name);
   }
+  // check idle animation name
+  auto actor_comp =
+    ecm.Component<ignition::gazebo::components::Actor>(entity);
+  for (auto idle_anim : _crowd_sim_interface->get_switch_anim_name())
```

Comment:
Nit, I'd probably make `idle_anim` a `std::unordered_set<std::string>` so you can just call the `find` function instead of having to iterate over the whole data structure.

Reply:
Emm, this `std::vector<std::string> _switch_anim_name` contains all the idle animation supported, like "idle", and "stand". In this for loop, we are iterate over {"idle", "stand"} on each Actor object, to find whether the actor sdf defines one of the idle animation. So if we use the unordered_set, we would probably still need to iterate the idle_animation.
BTW, this iteration only happens once for each actor when spawn and initialize the actor. If the idle animation is defined, this idle name will be stored, if not, idle name would be empty.

Reply:
Yea it's not really a bottleneck but it's just that if instead of being a vector it was an unordered_set the code would be quite a bit shorter (no need for a for loop, just a find call and a check on its result).

Reply:
Oops upon reading the code again I don't think it's possible to do it with a set, apologies for the noise!